airflow/providers/google/common/utils/id_token_credentials.py

```diff
@@ -0,0 +1,215 @@
+) -> Optional[google_auth_credentials.Credentials]:
+    """Gets credentials and project ID from the GCE Metadata Service."""
+    # Ping requires a transport, but we want application default credentials
+    # to require no arguments. So, we'll use the _http_client transport which
+    # uses http.client. This is only acceptable because the metadata server
+    # doesn't do SSL and never requires proxies.
+
+    # While this library is normally bundled with compute_engine, there are
+    # some cases where it's not available, so we tolerate ImportError.
+    try:
+        from google.auth import compute_engine
+        from google.auth.compute_engine import _metadata
+    except ImportError:
+        return None
+    from google.auth.transport import _http_client
+
+    if request is None:
+        request = _http_client.Request()
+
+    if _metadata.ping(request=request):
+        return compute_engine.IDTokenCredentials(
+            request, target_audience, use_metadata_identity_endpoint=True
+        )
+
+    return None
+
+
+def get_default_id_token_credentials(
+    target_audience: Optional[str], request: google.auth.transport.Request = None
+) -> google_auth_credentials.Credentials:
+    """Gets the default ID Token credentials for the current environment.
+
+    `Application Default Credentials`_ provides an easy way to obtain credentials to call Google APIs for
+    server-to-server or local applications.
+
+    .. _Application Default Credentials: https://developers.google.com\
+        /identity/protocols/application-default-credentials
+
+    :param target_audience: The intended audience for these credentials.
+    :type target_audience: Sequence[str]
+    :param request: An object used to make HTTP requests. This is used to detect whether the application
+            is running on Compute Engine. If not specified, then it will use the standard library http client
+            to make requests.
+    :type request: google.auth.transport.Request
+    :return the current environment's credentials.
+    :rtype google.auth.credentials.Credentials
+    :raises ~google.auth.exceptions.DefaultCredentialsError:
+        If no credentials were found, or if the credentials found were invalid.
+    """
+    checkers = (
+        lambda: _get_explicit_environ_credentials(target_audience),
+        lambda: _get_gcloud_sdk_credentials(target_audience),
+        lambda: _get_gce_credentials(target_audience, request),
+    )
+
+    for checker in checkers:
+        current_credentials = checker()
+        if current_credentials is not None:
+            return current_credentials
+
+    raise exceptions.DefaultCredentialsError(_HELP_MESSAGE)
+
+
+if __name__ == "__main__":
+    from google.auth.transport import requests
+
+    request_adaapter = requests.Request()
+
+    creds = get_default_id_token_credentials(target_audience=None)
+    creds.refresh(request=request_adaapter)
+    print(creds.token)
```

Comment:
Should it be removed?

Reply:
This is a helpful code because the ADC flow is very complex and sometimes you may need to manually check the tokens. At the beginning of the file, I added documentation that explains how to use it.